engine/src/main/java/org/hibernate/search/backend/SingularTermDeletionQuery.java

```diff
@@ -11,7 +11,7 @@
  *
  * @author Martin Braun
  */
-public final class SingularTermQuery implements DeletionQuery {
+public final class SingularTermDeletionQuery implements DeletionQuery {
```

Comment:
Is this meant to be a public type (i.e. to be referenced by users)? If so, that renaming would break existing usages. But if it is actually not meant to be a user-accessible class, it should be moved to some "impl" package.